metabase/metabase
# fix(sdk): missing css variables when rendering under a portal

Closes https://github.com/metabase/metabase/issues/44391

### Description

When SDK users are rendering the embedding SDK components (such as an interactive question) under a React portal (such as in a modal using their library of choice), the filter colors from the theme settings are not applied. In the screenshot below, you can notice that the filter color is not applied properly in (1).

![image](https://github.com/metabase/metabase/assets/4714175/ebb7e685-5434-4186-ae1f-35912115be3c)

This PR separates the components for injecting CSS variables to a public component (i.e. `PublicComponentStylesWrapper`) from the components injecting global sdk styles such as font-face declarations (i.e. `SdkGlobalStylesWrapper`), so we can provide the CSS variables in context of React portals.

### How to verify

- Go to the `fix-react-portal` branch under the Shoppy demo app
- Click on a product card, then click on the topmost chart
- The color should be correct

### Demo

![CleanShot 2567-06-20 at 21 12 11@2x](https://github.com/metabase/metabase/assets/4714175/edfc7aee-bde4-4d71-9a35-3bd053c56b9f)

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR - to add visual test or unit test next
